alacritty_terminal/src/term/cell.rs

```diff
@@ -104,8 +106,34 @@ impl Cell {
     #[inline(never)]
     pub fn clear_wide(&mut self) {
         self.flags.remove(Flags::WIDE_CHAR);
-        self.drop_extra();
         self.c = ' ';
+        if self.extra.as_ref().map_or(false, |extra| extra.hyperlink.is_none()) {
+            self.drop_extra();
+        }
```

Comment:
The zerowidth character should still be cleared. Pretty sure this will lead to incorrect behavior with zerowidth characters present.